### ISSUE TITLE
🐞 C.Set duplicate header

### ISSUE THREAD

**Issue by faultable** [gofiber/fiber#216]
**Fiber version/commit**
v1.8.2

**Issue description**
Just playing around with Fiber, and I have some problem here. In my case, I need to set custom headers dynamically. For example, I need to set `content-type` taken from database.

The code:

```go
func someRoute(c *fiber.Ctx) {

    for _, responseHeader := range response.Headers {
        c.Set(responseHeader.Key, responseHeader.Value)
    }

    c.Status(response.Status)
    c.Send(response.Body)
}
```

Let's say one of key from `responseHeader.key` is `content-type` and the value is `application/json`, the result is like this:

```bash
>  curl localhost:3000 --head

HTTP/1.1 200 OK
Content-Length: 16
Content-Type: text/plain; charset=utf-8
Content-Type: application/json
```

Probably we need to override the existing key?  Even though this is not a big deal, but this will lead to unpredictable problem.

**Expected behavior**
Override existing header in case the value is different.

```
> curl localhost:3000 --head

HTTP/1.1 200 OK
Content-Length: 16
Content-Type: application/json
```

**Steps to reproduce**
- Set custom header via `c.Set`
- Try to set existing header

**Code snippet**

```go
package main

import "github.com/gofiber/fiber"

// Header —
type Header struct {
    Key   string
    Value string
}

// Response —
type Response struct {
    Headers []Header
}

func helloHandler(c *fiber.Ctx) {
    var response = &Response{
        Headers: []Header{
            {
                Key:   "Content-Type",
                Value: "application/json",
            },
        },
    }

    for _, responseHeader := range response.Headers {
        c.Set(responseHeader.Key, responseHeader.Value)
    }

    c.Send("OKOK")
}

func main() {
    app := fiber.New()

    app.Get("/", helloHandler)
    app.Listen(3000)
}
```

**Comment by Fenny** [comment]
I changed two things regarding this problem for `v1.8.3`

- `c.Set()` uses fasthttp `SetCanonical` method to set headers, but assuming that key is in canonical form. `coNteNt-tyPe` will not be `normalized` into `Content-Type`  and caused duplicate problems.
- Currently `app.Get` routes are not trigger by `HEAD` requests ( your example `--head` ) and the default content-type is used.

Both these problems are fixed in `v1.8.3`
- `c.Set(key, value string)` will `normalize` header keys.
- `app.Get()` automatically called for `HEAD` method if no `app.Head()` was called for the path before `app.Get()`.

**Comment by faultable** [comment]
Thank you for confirming, can't wait!
